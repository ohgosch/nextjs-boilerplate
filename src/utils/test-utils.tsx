import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { ReactElement } from 'react';

import theme from 'visual/theme';

const customRender = (ui: ReactElement) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);

export { customRender as render, screen };
