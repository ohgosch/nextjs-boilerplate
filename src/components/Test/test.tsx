import { render, screen } from 'utils/test-utils';

import Test from '.';

describe('<Test />', () => {
  it('should render ...', () => {
    const { container } = render(<Test />);

    expect(screen.getByRole('heading', { name: /Test/i })).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
