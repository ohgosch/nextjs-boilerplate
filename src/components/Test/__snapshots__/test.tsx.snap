// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Test /> should render ... 1`] = `
<section
  class=""
>
  <h1>
    Test
  </h1>
  <svgrurl />
  <img
    src="/assets/logo-small.svg"
  />
</section>
`;
